### PR TITLE
Add Namespaces and NetworkServices for each example

### PR DESCRIPTION
**Pull request by NikitaSkrynnik**
Signed-off-by: Nikita Skrynnik <nikita.skrynnik@xored.com>

## Issue
https://github.com/networkservicemesh/deployments-k8s/issues/5436

### CODE REVIEW

File: examples/features/webhook/README.md [hunk 1]
```diff
@@ -63,62 +57,64 @@ spec:
...
 
-namespace: ${NAMESPACE}
+namespace: ns-webhook
 
 bases:
 - https://github.com/networkservicemesh/deployments-k8s/apps/nse-kernel?ref=3d1dcfe1de90681213c7f0006f25279bb4699966
 
 resources:
-- client.yaml
+- https://raw.githubusercontent.com/networkservicemesh/deployments-k8s/3d1dcfe1de90681213c7f0006f25279bb4699966/examples/features/webhook/netsvc.yaml
```
Comment (glazychev-art) [inline examples/features/webhook/README.md:82]: missed `client.yaml`

File: examples/features/dual-stack/Kernel2Wireguard2Kernel_dual_stack/README.md [hunk 2]
```diff
@@ -9,8 +9,7 @@ Forwarders are using the `wireguard` mechanism to connect with each other.
 
 Create test namespace:
 ```bash
-NAMESPACE=($(kubectl create -f https://raw.githubusercontent.com/networkservicemesh/deployments-k8s/3d1dcfe1de90681213c7f0006f25279bb4699966/examples/features/namespace.yaml)[0])
-NAMESPACE=${NAMESPACE:10}
+kubectl create ns ns-kernel2wireguard2kernel-dual-stack
```
Comment (anastasia-malysheva) [inline examples/features/dual-stack/Kernel2Wireguard2Kernel_dual_stack/README.md:12]: @NikitaSkrynnik wouldn't it be easier and more readable instead change just a value of NAMESPACE to ns-kernel2wireguard2kernel-dual-stack and use NAMESPACE later in the commands?

Comment (NikitaSkrynnik) [reply examples/features/dual-stack/Kernel2Wireguard2Kernel_dual_stack/README.md:12]: We decided to get rid of NAMESPACE env completely

File: examples/features/dual-stack/Kernel2Kernel_dual_stack/client.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    networkservicemesh.io: kernel://kernel2kernel-dual-stack/nsm-1
+spec:
+  containers:
+  - name: alpine
+    image: alpine:3.15.0
+    imagePullPolicy: IfNotPresent
+    stdin: true
+    tty: true
+  affinity:
+    podAntiAffinity:
```
Comment (glazychev-art) [inline examples/features/dual-stack/Kernel2Kernel_dual_stack/client.yaml:18]: should be `podAffinity`

File: examples/features/exclude-prefixes/README.md [hunk 4]
```diff
@@ -12,114 +12,50 @@ Make sure that you have completed steps from [basic](../../basic) or [memory](..
...
-        - name: nse
-          env:
-            - name: NSM_CIDR_PREFIX
-              value: 172.16.1.100/30
-      nodeName: ${NODE}
-EOF
+kubectl create ns ns-exclude-prefixes
 ```
 
+TODO: move it to kustomization file ?
```
Comment (glazychev-art) [inline examples/features/exclude-prefixes/README.md:18]: Do we need to fix this TODO?

My opinion - yes we should include it into kustomization file

File: examples/features/exclude-prefixes/patch-nsc.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+spec:
+  template:
+    spec:
+      containers:
+        - name: nsc
+          env:
+            - name: NSM_NETWORK_SERVICES
+              value: kernel://exclude-prefixes/nsm-1
+      affinity:
+        podAntiAffinity:
```
Comment (glazychev-art) [inline examples/features/exclude-prefixes/patch-nsc.yaml:15]: It was `podAffinity` previously...

File: examples/features/ipv6/Kernel2Kernel_ipv6/client.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    networkservicemesh.io: kernel://kernel2kernel-ipv6/nsm-1
+spec:
+  containers:
+  - name: alpine
+    image: alpine:3.15.0
+    imagePullPolicy: IfNotPresent
+    stdin: true
+    tty: true
+  affinity:
+    podAntiAffinity:
```
Comment (glazychev-art) [inline examples/features/ipv6/Kernel2Kernel_ipv6/client.yaml:18]: Please check affinity rules for all examples

File: examples/features/policy-based-routing/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ns-policy-based-routing
+
+resources:
+- client.yaml
```
Comment (glazychev-art) [inline examples/features/policy-based-routing/kustomization.yaml:8]: I think we should use the same notation in kustomize files - either relative path or full links.

In my opinion, relative paths are less prone to errors.

File: examples/features/scale-from-zero/README.md [hunk 8]
```diff
@@ -18,162 +18,81 @@ thus saving cluster resources (see step 14).
...
+Select nodes to deploy NSC and supplier:
 ```bash
 NODES=($(kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints }}{{ .metadata.name }} {{end}}{{end}}'))
 NSC_NODE=${NODES[0]}
 SUPPLIER_NODE=${NODES[1]}
 if [ "$SUPPLIER_NODE" == "" ]; then SUPPLIER_NODE=$NSC_NODE; echo "Only 1 node found, testing that pod is created on the same node is useless"; fi
 ```
 
-3. Create patch for NSC:
+TODO: move service registration to kustomization.yaml?
```
Comment (glazychev-art) [inline examples/features/scale-from-zero/README.md:34]: Please fix all TODOs

**Review by glazychev-art — COMMENTED** [review]
Good job!
Left a couple of comments

File: examples/heal/local-nsmgr-local-nse-memif/nse-after-death/kustomization.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ns-local-nsmgr-local-nse-memif
+
+bases:
+- https://github.com/networkservicemesh/deployments-k8s/examples/heal/local-nsmgr-local-nse-memif?ref=ba172266838eade1df97e2a8479fc73c3a9563c0
```
Comment (denis-tingaikin) [inline examples/heal/local-nsmgr-local-nse-memif/nse-after-death/kustomization.yaml:8]: Please use relative path

File: examples/heal/local-nse-death/nse-before-death/kustomization.yaml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ns-local-nse-death
+
+bases:
+- https://github.com/networkservicemesh/deployments-k8s/examples/heal/local-nse-death?ref=ba172266838eade1df97e2a8479fc73c3a9563c0
```
Comment (denis-tingaikin) [inline examples/heal/local-nse-death/nse-before-death/kustomization.yaml:8]: Please use relative path

File: examples/heal/local-nse-death/nse-after-death/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ns-local-nse-death
+
+bases:
+- https://github.com/networkservicemesh/deployments-k8s/examples/heal/local-nse-death?ref=ba172266838eade1df97e2a8479fc73c3a9563c0
```
Comment (denis-tingaikin) [inline examples/heal/local-nse-death/nse-after-death/kustomization.yaml:8]: Please use relative path

File: examples/floating_interdomain/spire/cluster1/kustomization.yaml [hunk 12]
```diff
@@ -23,4 +23,4 @@ configMapGenerator:
 
 
 bases:
-- https://github.com/networkservicemesh/deployments-k8s/examples/spire?ref=76c014f23bf7027dbd520ee957aa6c0135184dc7
+- https://github.com/networkservicemesh/deployments-k8s/examples/spire?ref=ba172266838eade1df97e2a8479fc73c3a9563c0
```
Comment (denis-tingaikin) [inline examples/floating_interdomain/spire/cluster1/kustomization.yaml:26]: Please use relative path

File: examples/floating_interdomain/spire/cluster3/kustomization.yaml [hunk 13]
```diff
@@ -22,4 +22,4 @@ configMapGenerator:
     - k8s-workload-registrar.conf
 
 bases:
-- https://github.com/networkservicemesh/deployments-k8s/examples/spire?ref=76c014f23bf7027dbd520ee957aa6c0135184dc7
+- https://github.com/networkservicemesh/deployments-k8s/examples/spire?ref=ba172266838eade1df97e2a8479fc73c3a9563c0
```
Comment (denis-tingaikin) [inline examples/floating_interdomain/spire/cluster3/kustomization.yaml:25]: Please use relative path

File: examples/floating_interdomain/usecases/FloatingKernel2Vxlan2Kernel/README.md [hunk 14]
```diff
@@ -38,7 +38,7 @@ kind: Kustomization
 namespace: ${NAMESPACE1}
 
 bases:
-- https://github.com/networkservicemesh/deployments-k8s/apps/nse-kernel?ref=76c014f23bf7027dbd520ee957aa6c0135184dc7
+- https://github.com/networkservicemesh/deployments-k8s/apps/nse-kernel?ref=0ac1af83b8560efa7d52ab7acb97bd7952429025
```
Comment (denis-tingaikin) [inline examples/floating_interdomain/usecases/FloatingKernel2Vxlan2Kernel/README.md:41]: Please use relative path

File: examples/floating_interdomain/usecases/FloatingKernel2Vxlan2Kernel/README.md [hunk 15]
```diff
@@ -108,7 +108,7 @@ kind: Kustomization
 namespace: ${NAMESPACE2}
 
 bases:
-- https://github.com/networkservicemesh/deployments-k8s/apps/nsc-kernel?ref=76c014f23bf7027dbd520ee957aa6c0135184dc7
+- https://github.com/networkservicemesh/deployments-k8s/apps/nsc-kernel?ref=0ac1af83b8560efa7d52ab7acb97bd7952429025
```
Comment (denis-tingaikin) [inline examples/floating_interdomain/usecases/FloatingKernel2Vxlan2Kernel/README.md:111]: Please use relative path

File: examples/floating_interdomain/usecases/FloatingKernel2Wireguard2Kernel/README.md [hunk 16]
```diff
@@ -81,7 +81,7 @@ kind: Kustomization
 namespace: ${NAMESPACE2}
 
 bases:
-- https://github.com/networkservicemesh/deployments-k8s/apps/nsc-kernel?ref=76c014f23bf7027dbd520ee957aa6c0135184dc7
+- https://github.com/networkservicemesh/deployments-k8s/apps/nsc-kernel?ref=0ac1af83b8560efa7d52ab7acb97bd7952429025
```
Comment (denis-tingaikin) [inline examples/floating_interdomain/usecases/FloatingKernel2Wireguard2Kernel/README.md:84]: Please use relative path

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**